sympy/polys/rings.py

```diff
@@ -2958,6 +3028,45 @@ def subresultants(self, g, x=None):
 
         return R
 
+    def free_variable(self):
+        """
+        Examples
+        ========
+
+        >>> from sympy import ZZ, ring
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 3*x**2 + 2*y**2 + 5*z**3 + 7*x**2*y
+        >>> p.free_variable()
+        {0, 1, 2}
```

Comment:
This method should be called `free_variables` or perhaps just `variables`: the name should indicate that it returns plural rather than singular.

With all of these methods it is worth checking what SageMath does because of there is a method there that does the same thing then it would be good to use the same name and ensure that the method here is consistent with that.